identity-account/src/account/account.rs

```diff
@@ -146,25 +148,36 @@ impl Account {
   ///
   /// See [`IdentityCreate`] to customize the identity creation.
   pub(crate) async fn create_identity(setup: AccountSetup, input: IdentitySetup) -> Result<Self> {
-    let command = CreateIdentity {
-      network: input.network,
-      method_secret: input.method_secret,
-      method_type: Self::key_to_method(input.key_type),
-    };
+    let (did_lease, state): (DIDLease, IdentityState) = create_identity(input, setup.storage.as_ref()).await?;
 
-    let snapshot = IdentitySnapshot::new(IdentityState::new());
+    let mut account = Self::with_setup(setup, ChainState::new(), state, did_lease).await?;
 
-    let (did, did_lease, events): (IotaDID, DIDLease, Vec<Event>) = command
-      .process(snapshot.identity().integration_generation(), setup.storage.as_ref())
-      .await?;
+    account.store_state().await?;
 
-    let commits = Self::commit_events(&did, &setup.config, setup.storage.as_ref(), &snapshot, &events).await?;
+    account.publish(false).await?;
 
-    let account = Self::with_setup(setup, did, did_lease).await?;
+    Ok(account)
+  }
 
-    account.publish_commits(snapshot, commits, true).await?;
+  /// Creates an [`Account`] for an existing identity, if it exists in the [`Storage`].
+  pub(crate) async fn load_identity(setup: AccountSetup, did: IotaDID) -> Result<Self> {
+    // Ensure the did exists in storage
+    let state = setup.storage.state(&did).await?.ok_or(Error::IdentityNotFound)?;
+    let chain_state = setup.storage.chain_state(&did).await?.ok_or(Error::IdentityNotFound)?;
 
-    Ok(account)
+    let did_lease = setup.storage.lease_did(&did).await?;
+
+    Self::with_setup(setup, chain_state, state, did_lease).await
+  }
+
+  // ===========================================================================
+  // Identity
+  // ===========================================================================
+
+  /// Resolves the DID Document associated with this `Account` from the Tangle.
+  pub async fn resolve_identity(&self) -> Result<IotaDocument> {
+    // Fetch the DID Document from the Tangle
+    self.client_map.resolve(self.did()).await.map_err(Into::into)
```

Comment:
Nit: redundant comment.
```suggestion
  /// Resolves the DID Document associated with this `Account` from the Tangle.
  pub async fn resolve_identity(&self) -> Result<IotaDocument> {
    self.client_map.resolve(self.did()).await.map_err(Into::into)
```